File: js/x2.js
"use strict";
/**
 * Example of attron and attroff and keyboard processing
 */

$(document).ready(function () {

    webcurses.initscr($("#container"));
    webcurses.printw("Type any character to see it in ");

    webcurses.attron('A_BOLD');
    webcurses.printw("bold\n");   // done    //console.timeEnd('init');
    webcurses.attroff('A_BOLD');

    webcurses.refresh();

    $(document).keypress(function (event) {
        console.log(event.which);
        if (webcurses.A_BOLD) {
            webcurses.attroff('A_BOLD');
        } else {
            webcurses.attron('A_BOLD');
        }
        webcurses.printw(String.fromCharCode(event.which));
        webcurses.refresh();
        event.preventDefault();
    });

});

File: c/x2.c
#include <ncurses.h>

int main()
{
    int ch;

    initscr();              // done
    raw();                  // no
    keypad(stdscr, TRUE);   // no
    noecho();               // no
    
    printw("Type any character to see it in bold\n");   // done
    ch = getch();           // no

    if(ch == KEY_F(1))
    {
        printw("F1 Key pressed");   // done
    }
    else
    {
        printw("The pressed key is ");
        attron(A_BOLD);     // todo
        printw("%c", ch);   // todo
        attroff(A_BOLD);    // todo
    }
    refresh();
    getch();
    endwin();

    return 0;
}
